### PR TITLE
New Rule: enforce-min-coverage-comments-sync

### PR DESCRIPTION
**Pull request by TSMMark**
# Why?

When using rule `enforce-min-coverage`, a common pattern in practice seems to be override the global eslint config value in several specific files, when the coverage is lower than required in that one file, but you are okay with the lower % for whatever reason.

That looks like:

```
// @flow
/* eslint "flowtype-errors/enforce-min-coverage": [2, 30] */

// ... all the file's code below
```

However, after years of doing this in certain files, we've come to a place where some files used to have low coverage %, but at some point, we have improved type-safety of underlying modules and consequently increased the coverage % of many files downstream.

So now we're left with many files with comments like `/* eslint "flowtype-errors/enforce-min-coverage": [2, 30] */`, when actually the coverage % could be much higher than that.

# New Rule Proposal

I have introduced a new rule called `enforce-min-coverage-comments-sync` which will update the per-file comments if the actual coverage % is greater than the `enforce-min-coverage` value by a configurable margin %.

 - If the file's actual coverage exceeds the local comment's % requirement, but is still below the global % requirement, the comment will be updated with a greater % value.
 - If the file's actual coverage exceeds the global % requirement, the comment will be removed entirely.

---

Let me know if you think this is a rule you would accept, and if so I can clean up this PR, rebase, whatever you want, lmk.

By the way, I ran this new rule on our medium-sized React codebase and it worked exactly as intended.

Thanks!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "eslint-plugin-flowtype-errors",
-  "version": "4.4.0",
+  "version": "4.5.0-pre0.4",
```
Comment (amilajack) [inline package.json:3]: This should be omitted

File: package.json [hunk 2]
```diff
@@ -58,6 +58,7 @@
     "eslint-plugin-vue": "^7.20.0",
     "execa": "^5.1.1",
     "flow-bin": "0.139.0",
+    "gitpkg": "^1.0.0-beta.2",
```
Comment (amilajack) [inline package.json:61]: Is this used?

File: src/index.js [hunk 3]
```diff
@@ -155,6 +155,23 @@ function createFilteredErrorRule(filter: (CollectOutputElement) => any): (contex
   };
 }
 
+const MIN_COVERAGE_DIRECTIVE_COMMENT_PATTERN = /(\s*eslint\s*['"]flowtype-errors\/enforce-min-coverage['"]\s*:\s*\[\s*(?:2|['"]error['"])\s*,\s*)(\d+)(\]\s*)/
```
Comment (amilajack) [inline src/index.js:158]: Break to new line. I'm curious why prettier isn't catching this.

**Review by amilajack — CHANGES_REQUESTED** [review]
Great stuff! Just a few minor comments

File: package.json (hunk 2 above)
Comment (TSMMark) [reply package.json:61]: Was using gitpkg to publish to a git ref in order to install in another codebase test the plugin there. Will clean it up !

File: src/index.js [hunk 4]
```diff
@@ -155,6 +155,24 @@ function createFilteredErrorRule(filter: (CollectOutputElement) => any): (contex
   };
 }
 
+const MIN_COVERAGE_DIRECTIVE_COMMENT_PATTERN =
+  /(\s*eslint\s*['"]flowtype-errors\/enforce-min-coverage['"]\s*:\s*\[\s*(?:2|['"]error['"])\s*,\s*)(\d+)(\]\s*)/
```
Comment (TSMMark) [inline src/index.js:159]: @amilajack I put a line break after `=`, but the pattern is still pretty long. AFAIK the only way to break this up further would be to use regexp constructor like this: https://stackoverflow.com/a/12317105/2696867

I personally prefer the regex literal, but lmk what you'd prefer

**Review by TSMMark — COMMENTED** [review]
@amilajack addressed comments, thanks for your really quick review!

File: src/index.js (hunk 4 above)
Comment (amilajack) [reply src/index.js:159]: This is fine, thanks! 😃

**Review by amilajack — APPROVED** [review]

**Comment by amilajack** [comment]
Published this to the `next` tag. Would appreciate if you could test it out before I publish to `latest`

```
npm i eslint-plugin-flowtype-errors@next
```

**Comment by amilajack** [comment]
Also windows builds seem to be failing

https://github.com/amilajack/eslint-plugin-flowtype-errors/runs/4216011150?check_suite_focus=true#step:5:126

Any ideas?

**Comment by TSMMark** [comment]
Just saw this. Will try to test `next` tag later tonight.

For the windows failure I can only guess something to do with EOL characters since the lines look identical to me

**Comment by TSMMark** [comment]
Confirmed `next` release works

**Comment by amilajack** [comment]
Published to `latest` 🥳

**Pull request merged**